src/components/input/code/views/inputView.jsx

```diff
@@ -0,0 +1,68 @@
+var React = require('react');
+
+module.exports = React.createClass({
+
+  intent: null,
+
+  propTypes: {
+    label: React.PropTypes.string,
+    type: React.PropTypes.oneOf(['text', 'password', 'email']),
+    placeholder: React.PropTypes.string,
+    defaultValue: React.PropTypes.string,
+    name: React.PropTypes.string,
+    id: React.PropTypes.string,
+    error: React.PropTypes.string,
+    disabled: React.PropTypes.bool,
+    readOnly: React.PropTypes.bool,
+    required: React.PropTypes.bool,
+    valid: React.PropTypes.bool,
+    validator: React.PropTypes.string,
+    errorMessage: React.PropTypes.string
+  },
+
+  getInitialState: function() {
+    return {
+      value: this.props.defaultValue || '',
+      error: null,
+      valid: true
+    };
+  },
+
+  getDefaultProps: function() {
+    return {
+      type: 'text',
+      disabled: false,
+      readOnly: false,
+      required: false,
+      errorMessage: null
+    };
+  },
+
+  handleChange: function(e) {
```

Comment:
Could you add a test to ensure this is working as expected?

I'd like to see a test where the component **isn't** passed an `errorMessage` and a test where the component **is** passed an `errorMessage`. Both of these tests should trigger an error and check the message rendered is what's expected.